extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -37,9 +40,50 @@ export class RemoteQueriesManager {
     private readonly storagePath: string,
     logger: Logger,
   ) {
+    super();
     this.analysesResultsManager = new AnalysesResultsManager(ctx, storagePath, logger);
     this.interfaceManager = new RemoteQueriesInterfaceManager(ctx, logger, this.analysesResultsManager);
     this.remoteQueriesMonitor = new RemoteQueriesMonitor(ctx, logger);
+
+    // Handle events from the query history manager
+    this.push(this.qhm.onDidAddQueryItem(this.handleAddQueryItem.bind(this)));
+    this.push(this.qhm.onDidRemoveQueryItem(this.handleRemoveQueryItem.bind(this)));
+    this.push(this.qhm.onWillOpenQueryItem(this.handleOpenQueryItem.bind(this)));
+  }
+
+  private async handleAddQueryItem(queryItem: QueryHistoryInfo) {
```

Comment:
A convention I've been following that I'd love to see (and we can discuss about whether we agree/want to enforce it with a linter) is to have public functions before private functions. I find that easier when I'm reading through code because I can easily see the public interface of a module.

I'd be great if we could do the same with these functions, but ignore me if you disagree/would cause too much merge pain.